### PR TITLE
added family id check before moving to pairing

### PR DESCRIPTION
**Pull request by amanCypherock**
https://app.clickup.com/t/37308523/CHI-2062

### CODE REVIEW

**Comment by vbcypherock** [comment]
Task linked: [CHI-2062 Add family id check after card auth in main](https://app.clickup.com/t/37308523/CHI-2062)

**Comment by ujjwal-cyph** [comment]
Add this change to `develop`

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]
Add this change to `develop`

File: src/level_four/core/controller/verify_card_controller.c [hunk 1]
```diff
@@ -199,8 +199,14 @@ void verify_card_controller()
 
     case VERIFY_CARD_SUCCESS:
         if (is_paired(tap_card_data.card_key_id) == -1) {
-            tap_card_data.desktop_control = false;    // moving to pairing; we don't want to convey anything to desktop
-            tap_card_take_to_pairing();
+            uint32_t device_family_id = U32_READ_BE_ARRAY(get_family_id());
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/level_four/core/controller/verify_card_controller.c:202]: Not a problem with this PR or ticket.
paired_card_family seems better but remember it still has to be valid

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**